lib/public/views/Runs/RunPerDataPass/RunsPerDataPassOverviewPage.js

```diff
@@ -68,22 +68,17 @@ export const RunsPerDataPassOverviewPage = ({
         displayOptions,
         dataPassId,
         sortModel,
+        cachedPdpBeamTypes,
     } = perDataPassOverviewModel;
 
     const activeColumns = {
         ...runsActiveColumns,
-        ...remoteRuns.match({
-            Success: (runs) => runs.some((run) => run.pdpBeamType === 'PbPb')
-                ? inelasticInteractionRateActiveColumnsForPbPb
-                : {},
-            Other: () => {},
-        }),
-        ...remoteRuns.match({
-            Success: (runs) => runs.some((run) => run.pdpBeamType === 'pp')
-                ? inelasticInteractionRateActiveColumnsForProtonProton
-                : {},
-            Other: () => {},
-        }),
+        ...cachedPdpBeamTypes.has('PbPb')
+            ? inelasticInteractionRateActiveColumnsForPbPb
+            : {},
+        ...cachedPdpBeamTypes.has('pp')
+            ? inelasticInteractionRateActiveColumnsForProtonProton
+            : {},
```

Comment:
If I understand correctly, you will add/remove columns dynamically depending on what rows you have? I think this will be very disturbing for the user for example when navigating from page to page, we should simply always have the columns

Reply:
No, because e.g. in case user has `PbPb` runs then filtering by `mu(INEL)` should not be available because it doesn't make sense at all, PbPb runs don't have this value, having this filter would be misleading